controllers/device_syncer.go

```diff
@@ -42,83 +42,70 @@ type DeviceSyncer struct {
 	deviceCli edgeCli.DeviceInterface
 	// syncing period in seconds
 	syncPeriod time.Duration
-	log        logr.Logger
+	Namespace  string
 }
 
 // NewDeviceSyncer initialize a New DeviceSyncer
-func NewDeviceSyncer(client client.Client,
-	logr logr.Logger,
-	periodSecs uint32, cfg *rest.Config) (DeviceSyncer, error) {
-	log := logr.WithName("syncer").WithName("Device")
-	coreMetaCliInfo := efCli.ClientURL{Host: "edgex-core-metadata", Port: 48081}
-	coreCmdCliInfo := efCli.ClientURL{Host: "edgex-core-command", Port: 48082}
-
-	nodePool, err := util.GetNodePool(cfg)
-	if err != nil {
-		return DeviceSyncer{}, err
-	}
-
+func NewDeviceSyncer(client client.Client, opts *options.YurtDeviceControllerOptions, nodepool string) (DeviceSyncer, error) {
```

Comment:
The function signature can be 
`func NewDeviceSyncer(client client.Client, opts *options.YurtDeviceControllerOptions) (DeviceSyncer, error) ` 
since nodePool can be retrieved from opts.

Reply:
NodePool has been retrieved from opts.